### ISSUE TITLE
item_file_download() incorrect error message

### ISSUE THREAD

**Issue by trashbirdecology** [USGS-R/sbtools#262]
Haven't had time to look deeper into this, but when running parameter "overwrite=FALSE" on `item_file_download()` I get the warning that "overwrite is FALSE". 
```
>sbtools::item_file_download(sb_id = sb_id,
+                    dest_dir = sb_dir, 
+                    overwrite=TRUE
+                    )
Error: Path exists and overwrite is FALSE
```

The path does exist but overwrite was TRUE.

**Comment by jread-usgs** [comment]
hmm.... seems the error message may be wrong in this? 

`overwrite_file` is the arg name, so `overwrite` will get passed into the `...` of the arguments and not used.

**Comment by trashbirdecology** [comment]
Aha, lol my bad

**Comment by trashbirdecology** [comment]
My bad ...

**Comment by trashbirdecology** [comment]
While you're here, though, I am trying to figure out why item_file_download isn't retrieving ALL files associated with the item.... I am not seeing, for example, file "RegionCodes.txt" for item "[5ea04e9a82cefae35a129d65](https://www.sciencebase.gov/catalog/item/5d65256ae4b09b198a26c1d7)".

```
# Define the item and create subdirectory for the item  -----------------------------------------------------------------
> sbtools::item_file_download(sb_id ="52b1dfa8e4b0d9b325230cd9")
> list.files(sb_dir)
 [1] "50-StopData.zip"                                            
 [2] "Completeness_Report_NABBS_Dataset_1966-2019.pdf"            
 [3] "MigrantNonBreeder.zip"                                      
 [4] "North American Breeding Bird Survey Dataset (1966-2019).xml"
 [5] "routes.zip"                                                 
 [6] "RunType.pdf"                                                
 [7] "SpeciesList.txt"                                            
 [8] "States"                                                     
 [9] "States.zip"                                                 
[10] "VehicleData.zip"                                            
[11] "Weather.zip"                                                

```

**Comment by dblodgett-usgs** [comment]
This should include: "RegionCodes.txt"

I see it in the item JSON: https://www.sciencebase.gov/catalog/item/5d65256ae4b09b198a26c1d7?format=json 

Will have a look.

```r
    sbtools::item_list_files("5ea04e9a82cefae35a129d65")
    #>                                                          fname     size
    #> 1              Completeness_Report_NABBS_Dataset_1966-2019.pdf   192730
    #> 2                                        MigrantNonBreeder.zip   206790
    #> 3                                                   routes.zip   120870
    #> 4                                                  RunType.pdf   528491
    #> 5                                                   States.zip 58065799
    #> 6                                              VehicleData.zip  2254423
    #> 7                                                  Weather.zip  2484574
    #> 8                                              SpeciesList.txt   281962
    #> 9  North American Breeding Bird Survey Dataset (1966-2019).xml   126828
    #> 10                                             50-StopData.zip 56006539
    #>                                                                                                                                        url
    #> 1  https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__5f%2F94%2Fc2%2F5f94c208e93224d95b62ef52cf77692da725a935
    #> 2  https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__bf%2Fe5%2Ff6%2Fbfe5f6834f85cc1e31edf67b5eb825b9abff5806
    #> 3  https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__5d%2Fca%2F74%2F5dca74b1e3e1c21f18443e8f27c38bf0e2b2a234
    #> 4  https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__83%2Fe4%2Fda%2F83e4dac260893015bbf3e501c8ef126c65258871
    #> 5  https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__38%2F0e%2F1d%2F380e1dd98a48aa48b9cf2efa25f74e07ebc797f8
    #> 6  https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__a9%2F97%2F2b%2Fa9972b26aaeb48bf9425ed21681312b4cc063a7c
    #> 7  https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__87%2Fb5%2F1d%2F87b51d999ae1ad18838aa60851e9bcff4498ac8d
    #> 8  https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__6f%2F16%2F1f%2F6f161fc7c7db1dcaf1259deb02d824700f280460
    #> 9  https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__b4%2F2f%2Fcf%2Fb42fcfe28a799db6e8c97200829ea1ebaccbf8ea
    #> 10 https://www.sciencebase.gov/catalog/file/get/5ea04e9a82cefae35a129d65?f=__disk__40%2Fe4%2F92%2F40e4925dde30ffd926b1b4d540b485d8a9a320ba
```
<sup>Created on 2020-09-09 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by dblodgett-usgs** [comment]
OK -- it's because we have a bunch of mixed up IDs.

```r
    sbtools::item_list_files("5d65256ae4b09b198a26c1d7")$fname
    #>  [1] "50-StopData.zip"                                               
    #>  [2] "BBSStrata.txt"                                                 
    #>  [3] "BCR.txt"                                                       
    #>  [4] "Completeness_Report-NABBS_Dataset_1966-2018_v2018_0.pdf"       
    #>  [5] "MigrantNonBreeder.zip"                                         
    #>  [6] "pkzipc.exe"                                                    
    #>  [7] "RegionCodes.txt"                                               
    #>  [8] "routes.zip"                                                    
    #>  [9] "RunProtocolID.txt"                                             
    #> [10] "RunType.pdf"                                                   
    #> [11] "SpeciesList.txt"                                               
    #> [12] "States.zip"                                                    
    #> [13] "VehicleData.zip"                                               
    #> [14] "Weather.zip"                                                   
    #> [15] "weathercodes.txt"                                              
    #> [16] "whatsnew.txt"                                                  
    #> [17] "File_and_Field_Definitions-NABBS_Dataset_1966-2018_v2018_0.pdf"
    #> [18] "MetaData-NABBS_Dataset_1966-2018_v2018_0.xml"
```
<sup>Created on 2020-09-09 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by trashbirdecology** [comment]
Mixed up IDs? New issue (I would like to be notified when this is resolved)

**Comment by jread-usgs** [comment]
I didn't end up finishing my other comment in response to this. It was something along the lines of "I don't tend to use `_download` to get all the files w/o specifying them. I either specify one, or specify a vector of files I want. Does your query work when you use output of `sbtools::item_list_files("5d65256ae4b09b198a26c1d7")$fname` as input to the `_download` call? Like Dave's example, but using that to be more explicit about which files you want. 

But yeah, this question is probably another issue (doesn't fit in the original question) which is probably why Dave closed it.

**Comment by dblodgett-usgs** [comment]
Sorry -- rushing through here. If you look closely at the various IDs above, they are all mixed up.

The link is to:
"5d65256ae4b09b198a26c1d7": https://www.sciencebase.gov/catalog/item/5d65256ae4b09b198a26c1d7

The link text is:
"5ea04e9a82cefae35a129d65": https://www.sciencebase.gov/catalog/item/5ea04e9a82cefae35a129d65

and the example code is:
"52b1dfa8e4b0d9b325230cd9": https://www.sciencebase.gov/catalog/item/52b1dfa8e4b0d9b325230cd9

When I tried to get the desired file from the correct item, everything works as expected, so I closed the issue.
